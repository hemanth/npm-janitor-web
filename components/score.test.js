import React from 'react'
import Score from './score'
import renderer from 'react-test-renderer'

it('renders correctly when module has errors', () => {
  let module = {
    errors: ['invalid name'],
    warnings: []
  }
  const tree = renderer.create(
    <Score module={module}/>
  ).toJSON()
  expect(tree).toMatchSnapshot()
})

it('renders correctly when score > 6 ', () => {
  let module = {
    errors: [],
    warnings: []
  }
  const tree = renderer.create(
    <Score module={module}/>
  ).toJSON()
  expect(tree).toMatchSnapshot()
})

it('renders correctly when score <= 6 ', () => {
  let module = {
    errors: [],
    warnings: ['warning 1', 'warning 2', 'warning 3', 'warning 4']
  }
  const tree = renderer.create(
    <Score module={module}/>
  ).toJSON()
  expect(tree).toMatchSnapshot()
})
