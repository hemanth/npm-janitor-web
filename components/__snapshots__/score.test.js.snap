// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly when module has errors 1`] = `
<div
  className="kUklBY"
/>
`;

exports[`renders correctly when score <= 6  1`] = `
<div
  className="jyRNRQ"
/>
`;

exports[`renders correctly when score > 6  1`] = `
<div
  className="huXapt"
/>
`;
